cumulus/parachains/common/src/xcm_config.rs

```diff
@@ -78,3 +79,85 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// Accepts an asset if it is a concrete asset from the system (Relay Chain or system parachain).
+pub struct ConcreteAssetFromSystem<AssetLocation>(PhantomData<AssetLocation>);
+impl<AssetLocation: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
+	for ConcreteAssetFromSystem<AssetLocation>
+{
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteAssetFromSystem asset: {:?}, origin: {:?}", asset, origin);
+		let is_system = match origin {
+			// The Relay Chain
+			MultiLocation { parents: 1, interior: Here } => true,
+			// System parachain
+			MultiLocation { parents: 1, interior: X1(Parachain(id)) } =>
+				ParaId::from(*id).is_system(),
+			// Others
+			_ => false,
+		};
+		matches!(asset.id, Concrete(id) if id == AssetLocation::get() && is_system)
```

Comment:
```suggestion
		matches!(asset.id, Concrete(id) if id == AssetLocation::get()) && is_system
```

Reply:
`is_system` is irrelevant to the `matches`, reads nicer if you leave it outside of the match guard.